discovery-infra/start_discovery.py

```diff
@@ -500,6 +540,42 @@ def set_hosts_roles(client, cluster, nodes_details, machine_net, tf, master_coun
                        update_roles=master_count > 1)
 
 
+def get_nodes_details(cluster_name, namespace, tf):
+    tf_folder = utils.get_tf_folder(cluster_name, namespace)
+    tf_vars = utils.get_tfvars(tf_folder)
+    networks_names = (
+        tf_vars['libvirt_network_name'],
+        tf_vars['libvirt_secondary_network_name'],
+    )
+    return utils.get_libvirt_nodes_from_tf_state(
+        network_names=networks_names,
+        tf_state=tf.get_state(),
+    )
+
+
+def set_agent_hostname(agent, nodes_details, is_ipv4):
```

Comment:
maybe some of the functions should be in some utility module for kubeapi functions